---
title: "ABDS User Maunal"
author: "Dongping Du"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

We developed the ABDS tool suite specifically for analyzing biologically diverse samples. Mechanism-integrated group-wise pre-imputation is designed to recruit signature genes that represent informative missingness, a cosine-based one-sample test is extended to detect downregulated signature genes, and a unified heatmap is designed to display multiple phenotypic groups. We discuss the methodological principles and demonstrate the effectiveness of three analytics tools, supported by comparative evaluations and biomedical showcases.

## Quick Start
You can install the latest version of ABDS from GitHub by
```{r, eval = FALSE}
devtools::install_github("niccolodpdu/ABDS")
```

## MGpI: Mechanism-integrated group-wise pre-imputation
Missing values associated with these genes are expected to have a group-specific mix of different missing mechanisms and cross-group uneven missing rates. Thus, using an overall missing rate for data quality control would be problematic and could adversely affect subsequent analyses. We propose a mechanism-integrated group-wise pre-imputation (MGpI) strategy that explicitly considers mixed missing mechanisms across different phenotypic groups.

In the following example, we will see how to apply MGpI to real-world gene expression data (GSE19380) with artificially introduced missing values from a mixed source of missingness - lower limit of detection (LLOD) and missing at random (MAR). 
```{r, eval = FALSE}
## Read-in

```



## eCOT: Extended Cosine-based One-sample Test
In addition to signature genes, a molecularly distinct group may also be characterized by features that are uniquely silent in the group of interest but in no others – so-called downregulated signature genes (DSGs).

## uniHM: Unified heatmap design for comparative display
A popular heatmap design for displaying differentially expressed genes is to standardize each gene separately – the expression levels are first centered and then normalized by standard deviation. To address the aforementioned drawbacks, we now propose an alternative heatmap design that can display both the differential pattern and referenced quality of SGs/DSGs.
